bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ErrorUtils.java

```diff
@@ -50,14 +51,28 @@
      * @return ballerina error
      */
     public static ErrorValue createInteropError(Throwable e) {
-        BMap<BString, Object> detailMap = new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL);
-        if (e.getMessage() != null) {
-            detailMap.put(ERROR_MESSAGE_FIELD, StringUtils.fromString(e.getMessage()));
-        }
-        if (e.getCause() != null) {
-            detailMap.put(ERROR_CAUSE_FIELD, createError(StringUtils.fromString(e.getCause().getClass().getName()),
-                                                         StringUtils.fromString(e.getCause().getMessage())));
+        MappingInitialValueEntry[] initialValues;
+        if (e.getMessage() != null && e.getCause() != null) {
```

Comment:
Shall we extract `e.getMessage()` and `e.getCause()` out to variables?

Reply:
Fixed